plugins/he-heading-renderer/src/renderer.tsx

```diff
@@ -0,0 +1,24 @@
+import * as React from 'react'
+import { HeHeadingPluginState } from '.'
+import { 
+    Document,
+    createDocumentIdentifier
+} from '@splish-me/editor-core-document'
+
+const renderChildren = (plugin) => {
+  let eid = createDocumentIdentifier((plugin as any).id);
+  return (<Document state={eid} initialState={plugin} key={(plugin as any).id}/>);
```

Comment:
Why do you create a new `DocumentIdentifier` on every render? Why not
```.ts
const renderChildren = (child) => {
  return <Document state={child} key={child.id} />
}
```
and let the editor handle persisting of the states of the children. Pretty sure your current approach breaks undo/redo.

Reply:
That's what I thought initially, but that does not work for me... It renders empty text(slate) plugins instead of the actual child...

Reply:
Shouldn't happen if you pass the state correctly. In this specific case, your plugin state should look something like this:
```.js
const pluginState = {
  "caption": "foobar",
  "content": [
    {
      "type": "@splish-me/editor-core/editable",
      "state": {} // This is the initial state of the document, i.e. currently `plugin`
    }
  ]
```
If that doesn't work for you, please upload the code and state that doesn't work and I'll debug it.

Reply:
Ah, this seems to work when I use `renderRenderer()` instead of `renderEditor()` in the storybook. So I'll have a look into initialization. 
Is there a way to load a complete plugin plugin tree from a JSON file or do I have to initialize the components recursively? Or in other words: What is the best way to load / store a complete document?

Reply:
You can (and should) store the document completely (that's also what the editor gives you on save and what is going to be persisted in the database at serlo.org). The workflow is:

- You give the editor the whole document (e.g. in the storybook by using `renderEditor` resp. `renderRenderer`
- The editor initializes the subdocuments for you, deserializes plugin states and passes the stuff to your plugins.
- When pressing save, the editor fetches the states of every subdocument, serializes plugin states and gives you the whole thing again as one JS object.

Reply:
If the editor initializes the subdocuments upon deserialization, where is their content stored? E.g. when I press save on a hint plugin with some text in its content, save yields:
```json
       ...
        "state": {
          "title": "Foobar",
          "content": {
            "$$typeof": "splish.document",
            "id": "899bb9f7-f0af-4a0b-8a2c-0be0c76a9950"
          }
        }
        ....
```
But where is the state of `899bb9f7-f0af-4a0b-8a2c-0be0c76a9950` saved? 

The editor does not seem to deserialize plugins in the state of plugins recursively. Creating a new `DocumentIdentifier` during rendering seemed to do that (at least made it work). So do I have to create `DocumentIdentifiers` for every child after deserialization?

Reply:
Should work correctly now, i.e. you shouldn't get any `$$typeof` stuff in the saved document anymore (and the editor should handle (de-)serialization correctly for you). Let me know if you still have issues with it :)